src/uavariant.cpp

```diff
@@ -488,24 +488,112 @@ UaString UaVariant::toFullString() const
 
 
 template<typename T>
-UaStatus UaVariant::toSimpleType( const UA_DataType* dataType, T* out ) const
+UaStatus UaVariant::toSimpleType( const UA_DataType* targetDataType, T* out ) const
```

Comment:
I don't understand the chosen logic in toSimpleType(). 
toSimpleType() is used to convert any UaVariant contents of a type which has fixed memory size.
Simple is not a synonym of Numeric (Numeric types are likely to be Simple though).
So with your changes, Simple types which are not Numeric will fail to be supported.

Proposal: the old branch with static_cast should be left as it was when neither source nor target type are numeric.

Reply:
Or, even simpler:

1. Source type == target type --> use the old branch,
2. Source type != target type but both are numeric -> try the numeric conversion
3. otherwise fail

Reply:
The client code of method toSimpleType is the implementations of
```
  UaStatus toBool( OpcUa_Boolean& value) const;
  UaStatus toInt16( OpcUa_Int16& value) const;
  UaStatus toUInt16( OpcUa_UInt16& value) const;
  UaStatus toInt32( OpcUa_Int32& value ) const;
  UaStatus toUInt32( OpcUa_UInt32& value ) const;
  UaStatus toInt64( OpcUa_Int64& value ) const;
  UaStatus toByte(OpcUa_Byte& value ) const;
  UaStatus toUInt64( OpcUa_UInt64& value ) const;
  UaStatus toFloat( OpcUa_Float&  value ) const;
  UaStatus toDouble( OpcUa_Double& value ) const;
```
it looks numeric.

Reply:
I still insist that Simple type != Numeric type even though the current coverage looks like it.
There are 26 built-in types out of which 10 are (a)numeric and 14 are (b)simple (=fixed-size). (a) implies (b) but not the other way. 
So I'd ask to keep the previous branch as the first possible choice in case source type == target type.